src/ol/source/VectorTile.js

```diff
@@ -300,6 +300,7 @@ class VectorTile extends UrlTile {
       this.tileKeysBySourceTileUrl_[sourceTileUrl].splice(index, 1);
       if (this.tileKeysBySourceTileUrl_[sourceTileUrl].length === 0) {
         delete this.tileKeysBySourceTileUrl_[sourceTileUrl];
+        this.sourceTiles_[sourceTileUrl].dispose();
```

Comment:
Not strictly related, but during debugging I found that we're not disposing of source tiles, so I added that.